### PR TITLE
Closes #534: Allow single input file

### PR DESCRIPTION
**Pull request by cdebourcy**
Widens range of inputs we can accept:
-allow either 1 input file or 2 paired input files
- allow fasta or fastq input

### CODE REVIEW

File: idseq_pipeline/commands/host_filtering_functions.py [hunk 1]
```diff
@@ -205,64 +208,72 @@ def run_star(fastq_file_1, fastq_file_2):
...
     # copy back to aws
     execute_command("aws s3 cp %s/%s %s/;" % (RESULT_DIR, PRICESEQFILTER_OUT1, SAMPLE_S3_OUTPUT_PATH))
-    execute_command("aws s3 cp %s/%s %s/;" % (RESULT_DIR, PRICESEQFILTER_OUT2, SAMPLE_S3_OUTPUT_PATH))
+    if len(input_fqs) == 2:
+        execute_command("aws s3 cp %s/%s %s/;" % (RESULT_DIR, PRICESEQFILTER_OUT2, SAMPLE_S3_OUTPUT_PATH))
 
-def run_fq2fa(input_fq_1, input_fq_2):
-    execute_command("sed -n '1~4s/^@/>/p;2~4p' <%s >%s/%s" % (input_fq_1, RESULT_DIR, FQ2FA_OUT1))
-    execute_command("sed -n '1~4s/^@/>/p;2~4p' <%s >%s/%s" % (input_fq_2, RESULT_DIR, FQ2FA_OUT2))
+def run_fq2fa(input_fqs):
```
Comment (boris-dimitrov) [inline idseq_pipeline/commands/host_filtering_functions.py]: I believe  `input_fq_1` is not defined here.

File: idseq_pipeline/commands/host_filtering_functions.py [hunk 2]
```diff
@@ -375,11 +387,8 @@ def run_stage1(lazy_run = True):
     fastq_files = execute_command_with_output("ls %s/*.%s" % (FASTQ_DIR, FILE_TYPE)).rstrip().split("\n")
 
     # Identify input files and characteristics
-    if len(fastq_files) <= 1:
-        return # only support paired reads for now
-    else:
-        fastq_file_1 = fastq_files[0]
-        fastq_file_2 = fastq_files[1]
+    if len(fastq_files) not in [1, 2]:
+        return # only support either 1 file or 2 (paired) files
```
Comment (boris-dimitrov) [inline idseq_pipeline/commands/host_filtering_functions.py]: Returning from the middle of a long function occasionally leads to hard to spot bugs being introduced in future.   Perhaps log a line when it happens.

File: idseq_pipeline/commands/host_filtering_functions.py (hunk 1 above)
Comment (cdebourcy) [reply idseq_pipeline/commands/host_filtering_functions.py]: Thanks!

**Comment by cdebourcy** [comment]
Thanks, fixed the critical errors

**Pull request merged**